aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -158,29 +168,117 @@ public void configureAndInitializeInApiModeWithoutApi() throws JSONException, Am
+            .put("syncIntervalInMinutes", 60);
+        AWSDataStorePlugin awsDataStorePlugin = new AWSDataStorePlugin(modelProvider, mockApiCategory);
+        awsDataStorePlugin.configure(dataStorePluginJson, context);
+        awsDataStorePlugin.initialize(context);
+
+        // Trick the DataStore since it's not getting initialized as part of the Amplify.initialize call chain
+        Amplify.Hub.publish(HubChannel.DATASTORE, HubEvent.create(InitializationStatus.SUCCEEDED));
+
+        assertRemoteSubscriptionsStarted();
+
+        Person person1 = createPerson("Test", "Dummy I");
+        Person person2 = createPerson("Test", "Dummy II");
+        Completable.fromSingle(single -> { // Save a record to local store
+            awsDataStorePlugin.save(person1, itemSaved -> {
+                assertNotNull(itemSaved.item().getId());
+                assertEquals(person1.getLastName(), itemSaved.item().getLastName());
+                single.onSuccess(true);
+            }, single::onError);
+        }).andThen(
+            Completable.fromSingle(single -> { // Verify the record has been saved
+                awsDataStorePlugin.query(Person.class, results -> {
+                    Person actualPerson = results.next();
+                    assertNotNull(actualPerson);
+                    assertFalse(results.hasNext()); // We should only have one result.
+                    assertEquals(person1, actualPerson);
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Clear the local store.
+                awsDataStorePlugin.clear(() -> {
+                    // Make sure the remote subscription operations were cancelled
+                    assertRemoteSubscriptionsCancelled();
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Save a new record to local store
+                awsDataStorePlugin.save(person2, itemSaved -> {
+                    assertEquals(person2.getLastName(), itemSaved.item().getLastName());
+                    // Check if the sync process restarted.
+                    assertRemoteSubscriptionsStarted();
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Verify the record has been saved
+                awsDataStorePlugin.query(Person.class, results -> {
+                    Person actualPerson = results.next();
+                    assertNotNull(actualPerson);
+                    assertFalse(results.hasNext()); //we should only have one result.
+                    assertEquals(person2, actualPerson);
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).blockingGet(OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
Does this construct work? It looks like you're making `assertFoo` statements on a different thread, which the jUnit test runner thread would not capture.

 If you put a `fail()` inside one these things, does the test runner thread catch it?